packages/flutter/lib/src/widgets/implicit_animations.dart

```diff
@@ -2108,3 +2108,95 @@ class _AnimatedPhysicalModelState extends AnimatedWidgetBaseState<AnimatedPhysic
     );
   }
 }
+
+/// Animated version of [FractionallySizedBox] which automatically transitions the
+/// child's size over a given duration whenever the given [widthFactor] or
+/// [heightFactor] changes, as well as the position whenever the given [alignment]
+/// changes.
+///
+/// For the animation, you can choose a [curve] as well as a [duration] and the
+/// widget will automatically animate to the new target [widthFactor] or
+/// [heightFactor].
+class AnimatedFractionallySizedBox extends ImplicitlyAnimatedWidget {
+  /// Creates a widget that sizes its child to a fraction of the total available
+  /// space that animates implicitly, and positions its child by an alignment
+  /// than animates implicitly.
+  ///
+  ///
+  ///
```

Comment:
Upon adding this example, I got the following error from one of the tests.

```
The following file does not have the right license header for dart files:
  /b/s/w/ir/x/w/flutter sdk/examples/api/lib/widgets/implicit_animations/animated_fractionally_sized_box.0.dart
The expected license header is:
// Copyright 2014 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.
...followed by a blank line.
```

I added it in so that it passes the test, but since im not a Flutter Author , I'm not sure how this makes sense

Reply:
You are a Flutter Author if you contribute code to Flutter. If you're wondering about copyright in the CLA, please take a look at the [CLA](https://cla.developers.google.com/about/google-corporate) under "Grant of Copyright License".

In any case, all source code needs to have this copyright header.